src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssBuffer.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurityNative
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct GssBuffer : IDisposable
+        {
+            internal UInt64 _length;
+            internal IntPtr _data;
+
+            internal int Copy(byte[] destination, int offset)
+            {
+                Debug.Assert(destination != null, "target destination cannot be null");
+                Debug.Assert(offset >= 0 && offset < destination.Length, "invalid offset " + offset);
+
+                if (_data == IntPtr.Zero || _length == 0)
+                {
+                    return 0;
+                }
+
+                int sourceLength = Convert.ToInt32(_length);
```

Comment:
If the intention here is to have it throw for a too-large value, a comment to that effect would be useful.